### ISSUE TITLE
Better matplotlib styling defaults

### ISSUE THREAD

**Issue by philippjfr** [ioam/holoviews#823]
Our default styling options were largely determined very early on when I fiddled with the matplotlib defaults a little bit and did not go through any detailed discussion. With matplotlib 2.0 (currently in beta) a lot of the default styling will change and I think that's a good time to also rethink some of our defaults. While matplotlib will be our main focus in this redesign, perhaps it's also worth thinking about whether we want our styling to be a bit more consistent between backends. I don't have any detailed suggestions yet but was hoping to kick off some discussion here, so please report what has always annoyed you about our defaults. I know the default matplotlib background color is something I definitely want to change.

**Comment by jlstevens** [comment]
I agree matplotlib 2.0 is a great excuse for us to also improve our defaults. :-)

**Comment by jbednar** [comment]
I agree as well; mpl 2.0 is the time, and improving consistency with Bokeh would also be great (as Bokeh is about to reach 1.0).  I'd personally be happy if all of our example notebooks came out reasonably similar between the two backends, after these updates.

For that to be meaningful I think we need to update our examples to use non-square aspects a lot more often, so that we notice behavior for long or tall plots better.

**Comment by philippjfr** [comment]
The matplotlib team have just uploaded the default styling changes for 2.0. You can see them [here](http://matplotlib.org/devdocs/users/dflt_style_changes.html).

**Comment by jlstevens** [comment]
I've assigned this to 1.8 and repurposed this issue to cover bokeh style changes as well.

My first proposed bokeh style change: make errorbars thicker my default:

<img src="https://cloud.githubusercontent.com/assets/890576/26598654/e1d9cc54-456d-11e7-8ecf-bdf8d714ba7f.png" width=250/>

**Comment by jlstevens** [comment]
I also think we should start adding concrete suggestions for style changes to this issue. E.g making the default colormap 'fire'?

**Comment by jbednar** [comment]
I think we should update to mpl 2.0 defaults in general; they look better to me, and they are what new users will get with mpl by default outside of hv.

**Comment by jbednar** [comment]
> My first proposed bokeh style change: make errorbars thicker by default:

I agree.  I also favor removing the grid lines by default, and making the default sizes comparable between the two backends.

**Comment by philippjfr** [comment]
I think we've addressed this for the most part. I'm sure there will be more tweaks we want to do so I'll reassign to 2.0.

**Comment by jlstevens** [comment]
> I think we've addressed this for the most part.

I agree! I just want to mention that #1610 was merged as well.
